source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -0,0 +1,396 @@
+    request.set_trigger_mode(triggerMode);
+    fgen::ConfigureTriggerModeResponse response;
+
+    ::grpc::Status status = GetStub()->ConfigureTriggerMode(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  }
+
+  fgen::ExportAttributeConfigurationBufferResponse export_attribute_configuration_buffer()
+  {
+    ::grpc::ClientContext context;
+    fgen::ExportAttributeConfigurationBufferRequest request;
+    request.mutable_vi()->set_id(GetSessionId());
+    fgen::ExportAttributeConfigurationBufferResponse response;
+
+    ::grpc::Status status = GetStub()->ExportAttributeConfigurationBuffer(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+    return response;
+  }
+
+  void import_attribute_configuration_buffer(fgen::ExportAttributeConfigurationBufferResponse export_buffer_response)
+  {
+    ::grpc::ClientContext context;
+    fgen::ImportAttributeConfigurationBufferRequest import_request;
+    import_request.mutable_vi()->set_id(GetSessionId());
+    auto exported_configuration = export_buffer_response.configuration();
+    import_request.mutable_configuration()->Add(exported_configuration.begin(), exported_configuration.end());
+    fgen::ImportAttributeConfigurationBufferResponse import_response;
+
+    ::grpc::Status status = GetStub()->ImportAttributeConfigurationBuffer(&context, import_request, &import_response);
+
+    EXPECT_TRUE(status.ok());
+    EXPECT_EQ(kfgenDriverApiSuccess, import_response.status());
+  }
+
+ private:
+  std::shared_ptr<::grpc::Channel> channel_;
+  std::unique_ptr<::nidevice_grpc::Session> driver_session_;
+  std::unique_ptr<fgen::NiFgen::Stub> nifgen_stub_;
+  std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+  std::unique_ptr<fgen::NiFgenLibrary> nifgen_library_;
+  std::unique_ptr<fgen::NiFgenService> nifgen_service_;
+  std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(nifgenDriverApiTest, PerformSelfTest_CompletesSuccessfuly)
+{
+  ::grpc::ClientContext context;
+  fgen::SelfTestRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  fgen::SelfTestResponse response;
+  ::grpc::Status status = GetStub()->SelfTest(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+  EXPECT_EQ(0, response.self_test_result());
+  EXPECT_LT(0, response.self_test_message().size());
+}
+
+TEST_F(nifgenDriverApiTest, PerformReset_CompletesSuccessfuly)
+{
+  ::grpc::ClientContext context;
+  fgen::ResetRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  fgen::ResetResponse response;
+  ::grpc::Status status = GetStub()->Reset(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  EXPECT_EQ(kfgenDriverApiSuccess, response.status());
+}
+
+TEST_F(nifgenDriverApiTest, SetAttributeViInt32_GetAttributeViInt32ReturnsSameValue)
```

Comment:
Nitpick: There are other driver tests that follow the styling you currently have, but arguably this might be better named and reworked as `SetAttributeViInt32_GetAttributeViInt32_ValueMatches` or similar. I'll defer to @reckenro on this.

Reply:
I agree, it seems like these tests should naturally follow the `Given_When_Then` / `Setup_Act_Verify` structure.

Practically, after you update the names to the format @SparkingSpork suggests I think you would just change the vertical spacing to follow. So this test starting at line 287 would switch from
```
::grpc::Status status = GetStub()->SetAttributeViInt32(&context, request, &response);

EXPECT_TRUE(status.ok()); 
EXPECT_EQ(kfgenDriverApiSuccess, response.status());  
ViInt32 get_attribute_value = get_int32_attribute(channel_name, attribute_to_set);  
EXPECT_EQ(expected_value, get_attribute_value);
```
becomes
```
::grpc::Status status = GetStub()->SetAttributeViInt32(&context, request, &response);
EXPECT_TRUE(status.ok()); 
EXPECT_EQ(kfgenDriverApiSuccess, response.status()); 

ViInt32 get_attribute_value = get_int32_attribute(channel_name, attribute_to_set);  

EXPECT_EQ(expected_value, get_attribute_value);
```

Reply:
Made this change at required places.